core/client/app/controllers/setup/two.js

```diff
@@ -16,29 +15,46 @@ export default Ember.Controller.extend(ValidationEngine, {
     notifications: Ember.inject.service(),
     application: Ember.inject.controller(),
 
-    gravatarUrl: Ember.computed('email', function () {
-        var email = this.get('email'),
-            size = this.get('size');
-
-        return 'http://www.gravatar.com/avatar/' + md5(email) + '?s=' + size + '&d=blank';
-    }),
-
-    userImage: Ember.computed('gravatarUrl', function () {
-        return this.get('image') || this.get('gravatarUrl');
-    }),
-
-    userImageBackground: Ember.computed('userImage', function () {
-        return 'background-image: url(' + this.get('userImage') + ')';
-    }),
-
     // ValidationEngine settings
     validationType: 'setup',
 
+    /**
+     * Uploads the given data image, then sends the changed user image property to the server
+     * @param  {Object} user User object, returned from the 'setup' api call
+     * @return {Ember.RSVP.Promise} A promise that takes care of both calls
+     */
+    sendImage: function (user) {
+        var self = this,
+            image = this.get('image');
+
+        return new Ember.RSVP.Promise(function (resolve, reject) {
+            image.formData = {};
+            image.submit()
+                .success(function (response) {
+                    user.image = response;
+                    ajax({
+                        url: self.get('ghostPaths.url').api('users', user.id.toString()),
+                        type: 'PUT',
+                        data: {
+                            users: [user]
+                        }
+                    }).then(function () {
+                        resolve();
+                    }).catch(function (resp) {
+                        reject(resp);
+                    });
```

Comment:
You can just put the resolve/reject directly in your then/catch 
```js
.then(resolve)
.catch(reject);
```
(right? am I crazy?)